annil/src/main.rs

```diff
@@ -40,66 +40,77 @@ pub struct AppState {
     etag: RwLock<String>,
 }
 
-fn init_metadata(config: &Config) -> anyhow::Result<PathBuf> {
+struct LazyDb {
```

Comment:
What about `Lazy<RepoDatabaseRead>`?

Reply:
We cannot get an owned `RepoDatabaseRead` with `Lazy<RepoDatabaseRead>`  but a reference.

Besides, because the initialization can return errors, we can only have a `Lazy<Result<RepoDatabaseRead>>`  without using panics, which is hard to handle.